packages/mdc-toolbar/adapter.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import {SpecificEventListener} from '@material/base/types';
+import {ToolbarEventDetail} from './types';
+
+/**
+ * Defines the shape of the adapter expected by the foundation.
+ * Implement this adapter for your framework of choice to delegate updates to
+ * the component in your framework of choice. See architecture documentation
+ * for more details.
+ * https://github.com/material-components/material-components-web/blob/master/docs/code/architecture.md
+ */
+export interface MDCToolbarAdapter {
+  hasClass: (className: string) => boolean;
+  addClass: (className: string) => void;
+  removeClass: (className: string) => void;
+  registerScrollHandler: (handler: SpecificEventListener<'scroll'>) => void;
+  deregisterScrollHandler: (handler: SpecificEventListener<'scroll'>) => void;
+  registerResizeHandler: (handler: SpecificEventListener<'resize'>) => void;
+  deregisterResizeHandler: (handler: SpecificEventListener<'resize'>) => void;
+  getViewportWidth: () => number;
+  getViewportScrollY: () => number;
+  getOffsetHeight: () => number;
+  getFirstRowElementOffsetHeight: () => number;
+  notifyChange: (evtData: ToolbarEventDetail) => void;
+  setStyle: (property: string, value: string) => void;
+  setStyleForTitleElement: (property: string, value: string) => void;
+  setStyleForFlexibleRowElement: (property: string, value: string) => void;
+  setStyleForFixedAdjustElement: (property: string, value: string) => void;
+}
+
+export default MDCToolbarAdapter;
```

Comment:
I figured we should make all of our exports consistent, so that we don't need to remember which ones need a legacy `default` export and which ones don't.